### ISSUE TITLE
beam.smp seems to hang with 100% CPU when `mix dialyzer` on elixir 1.7

### ISSUE THREAD

**Issue by seeekr** [jeremyjh/dialyxir#229]
Console output:

```
$ mix dialyzer
Compiling 8 files (.ex)
Finding suitable PLTs
Checking PLT...
[:asn1, :certifi, :combine, :compiler, :cowboy, :cowlib, :crypto, :eex, :elixir, :gettext, :hackney, :httpotion, :ibrowse, :idna, :kernel, :logger, :metrics, :mime, :mimerl, :phoenix, :phoenix_pubsub, :plug, :poison, :porcelain, :public_key, :ranch, :retry, :ssl, :ssl_verify_fun, :stdlib, :timex, :tzdata, :unicode_util_compat]
Looking up modules in dialyxir_erlang-21.0.3_elixir-1.7.0_deps-dev.plt
Looking up modules in dialyxir_erlang-21.0.3_elixir-1.7.0.plt
Finding applications for dialyxir_erlang-21.0.3_elixir-1.7.0.plt
Finding modules for dialyxir_erlang-21.0.3_elixir-1.7.0.plt
Checking 184 modules in dialyxir_erlang-21.0.3_elixir-1.7.0.plt
Adding 230 modules to dialyxir_erlang-21.0.3_elixir-1.7.0.plt
```

Seems to hang forever, let it run for > 10 minutes and no progress.

MacOS 10.13.x, `{:dialyxir, "~> 1.0.0-rc.3", only: [:dev], runtime: false}`

**Comment by tmepple** [comment]
It's probably the same as https://github.com/elixir-lang/elixir/issues/7989 which is looking like a dialyzer issue at this point to be reported upstream.

**Comment by PragTob** [comment]
and already being fixed in elixir-lang/elixir#7993 - thanks community!
